common/app/services/newsletters/GroupedNewslettersAgent.scala

```diff
@@ -0,0 +1,57 @@
+package services.newsletters
+
+import com.gu.Box
+import common.GuLogging
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class GroupedNewslettersAgent(newsletterApi: NewsletterApi) extends GuLogging {
+
+  val emptyGroupedNewslettersResponse = GroupedNewslettersResponse(
+    GroupedNewsletterResponse("News roundups", Nil),
+    GroupedNewsletterResponse("News by topic", Nil),
+    GroupedNewsletterResponse("Features", Nil),
+    GroupedNewsletterResponse("Fport", Nil),
+    GroupedNewsletterResponse("Culture", Nil),
+    GroupedNewsletterResponse("Lifestyle", Nil),
+    GroupedNewsletterResponse("Comment", Nil),
+    GroupedNewsletterResponse("Work", Nil),
+    GroupedNewsletterResponse("From the papers", Nil),
+  )
+
+  private val groupedNewslettersAgent =
+    Box[Either[String, GroupedNewslettersResponse]](Right(emptyGroupedNewslettersResponse))
+
+  def refresh()(implicit ec: ExecutionContext): Future[Either[String, GroupedNewslettersResponse]] = {
+    log.info("Refreshing Grouped Newsletters for round up page.")
+
+    val groupedNewslettersQuery = newsletterApi.getGroupedNewsletters()
+
+    groupedNewslettersQuery.flatMap { newsletters =>
+      log.info(s"*** ${newsletters.toString}")
+      groupedNewslettersAgent.alter(newsletters match {
+        case Right(response) =>
+          log.info("Successfully refreshed Grouped Newsletters cache.")
+          Right(response)
+        case Left(err) =>
+          log.error(s"Failed to refresh Grouped Newsletters cache: $err")
+          Left(err)
+      })
```

Comment:
I vaguely recall there being some abstractions over agents that do some of this work for you, but that might've been specific to commercial agents or something